op-service/sources/prefetching_eth_client_test.go

```diff
@@ -0,0 +1,302 @@
+package sources
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+// MockEthClient is a mock of EthClient used for testing.
+type MockEthClient struct {
+	mock.Mock
+}
+
+func (m *MockEthClient) InfoByNumber(ctx context.Context, number uint64) (eth.BlockInfo, error) {
+	args := m.Called(ctx, number)
+	return args.Get(0).(eth.BlockInfo), args.Error(1)
+}
+
+// // MockRPC is a mock of the RPC client, similar to mockRPC but tailored for PrefetchingEthClient.
+// // Add any additional methods that might be required for testing.
+
+// // ... [MockRPC implementation] ...
+
+// var _ client.RPC = (*MockRPC)(nil)
+
```

Comment:
Consider a test object where you pass the endpoint/request function (`InfoByNumber`, etc.) - then you can have one or a handful of tests that repeat the same assertions against the various endpoints (`InfoByNumber`, `InfoByHash`, and so on)

Reply:
Can also supply prefetching range

Reply:
Just one test function that gets called a bunch of different ways (with `prefetchingRange` = `0, 1, 5`, etc; with different endpoints as well)